docs/multi-cloud-deployment-guide.md

```diff
@@ -0,0 +1,489 @@
+cd ..
+
+# Wait until the cluster shows the correct number of nodes, usually about a minute
+
+git clone https://github.com/Azure/fleet-networking
+cd fleet-networking
+./hack/membership/joinMC.sh v0.14.8 v0.3.8 hub $clusterName
+cd ..
+```
+
+These commands also will work to add clusters from other cloud providers 
+
+## Installing Operators and Dependencies
+
+1. Create necessary namespaces on the hub cluster:
+
+```bash
+kubectl create namespace cnpg-system || true
+```
+
+2. Install CloudNativePG operator on the hub cluster:
+
+```bash
+helm repo add cnpg https://cloudnative-pg.github.io/charts
+helm repo update
+helm install cnpg-operator cnpg/cloudnative-pg --namespace cnpg-system --wait
+```
+
+3. Install cert-manager on each cluster:
+
+```bash
+kazure apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.17.1/cert-manager.yaml
+konprem apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.17.1/cert-manager.yaml
+```
+
+3. Install sidecar injector on the hub:
+```bash
+cat <<EOF > sidecar-injector.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    cnpg.io/pluginClientSecret: helloworld-client-tls
+    cnpg.io/pluginPort: "9090"
+    cnpg.io/pluginServerSecret: helloworld-server-tls
+  labels:
+    app: hello-world
+    cnpg.io/pluginName: cnpg-i-hello-world.cloudnative-pg.io
+  name: hello-world
+  namespace: cnpg-system
+spec:
+  ports:
+  - port: 9090
+    protocol: TCP
+    targetPort: 9090
+  selector:
+    app: hello-world
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: hello-world
+  name: hello-world
+  namespace: cnpg-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: hello-world
+  strategy: {}
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: hello-world
+    spec:
+      containers:
+      - args:
+        - plugin
+        - --server-cert=/server/tls.crt
+        - --server-key=/server/tls.key
+        - --client-cert=/client/tls.crt
+        - --server-address=:9090
+        image: pgcosmoscontroller.azurecr.io/cnpg-plugin:v15
+        name: cnpg-i-hello-world
+        ports:
+        - containerPort: 9090
+          protocol: TCP
+        resources: {}
+        volumeMounts:
+        - mountPath: /server
+          name: server
+        - mountPath: /client
+          name: client
+      volumes:
+      - name: server
+        secret:
+          secretName: helloworld-server-tls
+      - name: client
+        secret:
+          secretName: helloworld-client-tls
+EOF
+
+kubectl apply -f sidecar-injector.yaml
+```
+
+4. Install the DocumentDB operator on the hub:
+
+```bash
+VERSION=36  # Replace with your current version
+helm package documentdb-chart --version 0.0.${VERSION}
+helm install documentdb-operator ./documentdb-operator-0.0.${VERSION}.tgz --namespace documentdb-operator-ns 
+```
+
+5. Deploy certificates on each cluster:
```

Comment:
Maybe we need to mention how many certificates we need and where they are used. And then mention we are using self-signed certs here?